ddtrace/contrib/aws_lambda/patch.py

```diff
@@ -32,21 +32,25 @@ def _crash_flush(_, __):
 
 def _handle_signal(sig, f):
     """
-    Wraps the given signal with a previously defined, if exists.
+    Wraps the given signal with a previously defined signal, if
+    it's callable and is different than the function `f` to handle.
+    Else, it doesn't wrap and just sets the function to handle as `f`.
     This to avoid our signals overriding existing ones.
 
-    Returns the handler of the wrapped signal.
+    Returns a signal of type `sig` with function `_f`
     """
-    old_signal = None
-    if callable(signal.getsignal(sig)):
-        old_signal = signal.getsignal(sig)
+    old_signal = signal.getsignal(sig)
+    if not callable(old_signal) or old_signal == f:
+        old_signal = None
 
     def wrap_signals(*args, **kwargs):
         if old_signal is not None:
             old_signal(*args, **kwargs)
         f(*args, **kwargs)
 
-    return signal.signal(sig, wrap_signals)
+    _f = wrap_signals if old_signal else f
```

Comment:
```suggestion
```
Let me know if this is functionally the same. To me, this is easier to follow.

Reply:
It worked as expected in the tests, and yeah, agree, easier to read!